docs/fleet/configure-a-fleet.md

```diff
@@ -78,7 +78,262 @@ For an example of adding a fragment to a machine, see the [Add a Rover Fragment
 ## Modify the config of a machine that uses a fragment
 
 When you modify a fragment, those changes are pushed to all machines that use that fragment.
-If you need to modify the config of just one machine that uses a fragment you can do the following:
+If you need to modify the config of just one machine that uses a fragment you have two options:
+
+- Use `fragment_mods` in your machine's config to overwrite certain fields of the fragment.
+- Copy and paste the contents of the fragment, remove the link to the fragment itself, then modify the config as needed.
+  - If you use this method, future updates to the fragment _will not_ be automatically pushed to your machine.
+
+### Use `fragment_mods`
+
+You can modify fragment fields in your machine's raw JSON config by using [update operators](https://www.mongodb.com/docs/manual/reference/operator/update/positional/#---update-).
+Viam supports all update operators except for `$setOnInsert`, `$`, `$[]`, and `$[<identifier>]`.
+
+1. Navigate to your machine's **Config** tab.
+2. Switch to **Raw JSON** mode.
+3. Add a top-level section called `"fragment_mods"` (alongside the other top-level sections like `"components"` and `"fragments"`):
+
+{{< tabs >}}
+{{% tab name="Template" %}}
+
+```json {class="line-numbers linkable-line-numbers"}
+  "fragment_mods": [
+    {
+      "fragment_id": "<YOUR FRAGMENT ID>",
+      "mods": [
+        {
+          <INSERT YOUR MODS HERE>
+        }
+      ]
+    }
+  ],
+```
+
+{{% /tab %}}
+{{% tab name="Full example" %}}
+This example assumes the fragment with ID `abcd7ef8-fa88-1234-b9a1-123z987e55aa` contains a motor configured with `"name": "motor1"`.
+
+```json {class="line-numbers linkable-line-numbers"}
+{
+  "components": [],
+  "fragment_mods": [
+    {
+      "fragment_id": "abcd7ef8-fa88-1234-b9a1-123z987e55aa",
+      "mods": [
+        {
```

Comment:
looking back it this - I think when the command is the same, it shouldn't be repeated - just have it as:

```

"$set": {
"components.motor1.attributes.max_rpm": 1818,
"components.motor1.attributes.pins.a": 30,
"components.motor1.attributes.board": "local"
}

```

Unless you were able to get it to work this way . Going to test this right now though to double check.

Reply:
Yeah I think they may have to be stacked if the command is the same within the same fragment_mods entry - there's a duplicate key error in the json and it discards the second.


https://github.com/viamrobotics/docs/assets/51691446/f2b0f49c-1c7b-4a30-b52d-820f1c5eb6e4

Reply:
Oh interesting! That looks a lot cleaner. I didn't try combining them--I only tried it repeated, which did work also!

Reply:
oh interesting 🤔 I wonder why mine didn't that's odd